icgcget/clients/download_client.py

```diff
@@ -78,9 +78,18 @@ def _run_command(self, args, parser, env=None):
         env['PATH'] = '/usr/local/bin:' + env['PATH']  # virtalenv compatibility
         try:
             output = ''
+
             process = subprocess.Popen(args, bufsize=0,  stdout=subprocess.PIPE, stderr=subprocess.STDOUT, env=env)
             if self.session:
                 self.session['subprocess'].append(process.pid)
+                if cidfile_name:
+                    while True:
```

Comment:
This could most definitely lead to an infinite loop. If the file is never created this will continue to loop forever. 

You should have a maximum number of attempts (for example 5) and sleep between the attempts. Also log the attempts.